pkg/controllers/rollout/controller.go

```diff
@@ -563,6 +562,16 @@ func (r *Reconciler) calculateRealTarget(crp *fleetv1beta1.ClusterResourcePlacem
 // A binding with not trackable resources is considered ready if the binding's current spec has been available before
 // the ready cutoff time.
 func isBindingReady(binding *fleetv1beta1.ClusterResourceBinding, readyTimeCutOff time.Time) (time.Duration, bool) {
+	// the binding is ready if the diff report has been reported
```

Comment:
Hi Ryan! Aside from the `isBindingReady` function we also have a `HasBindingFailed` function. Before the update the two functions are mutually exclusive, i.e., if `isBindingReady` returns true, `HasBindingFailed` would always return false. With the new edit this no longer seems to be stand.

Reply:
Which brings up a situation that, for a bound binding with the ReportDiff apply strategy, it would be considered as a ready binding (because the `DiffReported` condition is true) and a failed binding at the same (because it has a false `Applied` conditon and no `Available` condition).